internal/pkg/agent/application/upgrade/artifact/download/snapshot/verifier.go

```diff
@@ -41,8 +41,9 @@ func NewVerifier(log *logger.Logger, config *artifact.Config, pgp []byte, versio
 }
 
 // Verify checks the package from configured source.
-func (v *Verifier) Verify(a artifact.Artifact, version string, skipDefaultPgp bool, pgpBytes ...string) error {
-	return v.verifier.Verify(a, version, skipDefaultPgp, pgpBytes...)
+func (v *Verifier) Verify(a artifact.Artifact, version agtversion.ParsedSemVer, skipDefaultPgp bool, pgpBytes ...string) error {
+	strippedVersion := agtversion.NewParsedSemVer(version.Major(), version.Minor(), version.Patch(), version.Prerelease(), "")
```

Comment:
Same as in the download case, what happens if we don't strip the build metadata?

(I also worry about having to duplicate this across two files but that is probably too much refactoring to pull into this change)

Reply:
If we leave the build metadata in, no existing snapshot package file name will match, even the ones that currently work for the upgrade to a specific snapshot build, for example looking at https://artifacts-api.elastic.co/v1/versions/8.12.0-SNAPSHOT/builds/8.12.0-7d5211d6

```json
 "elastic-agent-8.12.0-SNAPSHOT-linux-x86_64.tar.gz": {
            "url": "https://snapshots.elastic.co/8.12.0-7d5211d6/downloads/beats/elastic-agent/elastic-agent-8.12.0-SNAPSHOT-linux-x86_64.tar.gz",
            "sha_url": "https://snapshots.elastic.co/8.12.0-7d5211d6/downloads/beats/elastic-agent/elastic-agent-8.12.0-SNAPSHOT-linux-x86_64.tar.gz.sha512",
            "asc_url": "https://snapshots.elastic.co/8.12.0-7d5211d6/downloads/beats/elastic-agent/elastic-agent-8.12.0-SNAPSHOT-linux-x86_64.tar.gz.asc",
            "type": "tar",
            "architecture": "x86_64",
            "os": [
              "linux"
            ]
          },
```
We can avoid stripping the metadata only once the downloaders become a lot smarter at looking up versions but that is a big change

The duplication is due to the way the code is structured and I wouln't want

Reply:
As noted above, let's leave this alone.